lib/ansible/modules/database/mysql/mysql_db.py

```diff
@@ -346,81 +372,65 @@ def main():
     changed = False
     if not os.path.exists(config_file):
         config_file = None
-    if db_exists(cursor, db):
-        if state == "absent":
-            if module.check_mode:
-                module.exit_json(changed=True, db=db_name)
+
+    existence_list = []
+    non_existence_list = []
+    for each_database in db:
+        if db_exists(cursor, [each_database]):
+            existence_list.append(each_database)
+        else:
+            non_existence_list.append(each_database)
+
+    if state == "absent":
+        if module.check_mode:
+            module.exit_json(changed=bool(existence_list), db=db_name, db_list=db)
+        try:
+            changed = db_delete(cursor, existence_list)
+        except Exception as e:
+            module.fail_json(msg="error deleting database: %s" % to_native(e))
+        module.exit_json(changed=changed, db=db_name, db_list=db)
+    elif state == "present":
+        if module.check_mode:
+            module.exit_json(changed=bool(non_existence_list), db=db_name, db_list=db)
+        changed = False
+        if non_existence_list:
             try:
-                changed = db_delete(cursor, db)
+                changed = db_create(cursor, non_existence_list, encoding, collation)
```

Comment:
You want `changed` to be `True` here if *at least* one DB was created, not if *all* DBs in `non_existence_list` were created (and there has been at least one).

Reply:
fixed